API/Parser/Parser.cs

```diff
@@ -64,43 +64,43 @@ public static class Parser
         /// <summary>
         /// Recognizes the Special token only
         /// </summary>
-        private static readonly Regex SpecialTokenRegex = new Regex(@"SP\d+",
+        private static readonly Regex SpecialTokenRegex = new Regex(@"(SP|HS)\d+",
             MatchOptions, RegexTimeout);
 
+        private const string seriesExpr = $@"(?<Series>{balanced}?)";
+        private const string volumeExpr = @"(?<Volume>\d+(\.\d)?(-\d+(\.\d)?)?)";
+        private const string chapterExpr = @"(?<Chapter>\d+(\.\d)?(-\d+(\.\d)?)?)";
+        private const string balancedParen = @"(?:[^()]|(?<open>\()|(?<-open>\)))+?(?(open)(?!))";
+        private const string balancedBrackets = @"(?:[^\[\]]|(?<openbracket>\[)|(?<closebracket>-<openbracket>\]))+?(?(openbracket)(?!))";
+        // non greedy matching of a string where parenthesis and brackets are balanced
+        private const string balanced = @"(?:[^()\[\]]|(?<open>\()|(?<-open>\))|(?<openbracket>\[)|(?<-openbracket>\]))*?(?(openbracket)(?!))(?(open)(?!))";
+        // French Volume: tome, t, t. -> volume
+        private const string volumePrefixes = @"(v|t|(v\.|vol\.?|volume|t\.|tome)\s?)";
+        // n°,n. -> numero sign
+        private const string chapterPrefixes = @"(c|chapter[-\s]+|issue\s?#?|#|(n°|n\.|Ep\.?)[-\s]*)";
+
 
         private static readonly Regex[] MangaVolumeRegex = new[]
         {
-            // Dance in the Vampire Bund v16-17
+            // vol_001-1.cbz for MangaPy default naming convention
             new Regex(
-                @"(?<Series>.*)(\b|_)v(?<Volume>\d+-?\d+)( |_)",
+                @"^(vol\s)(?<Volume>\d+(\.\d)?)-\d+$",
```

Comment:
I'm not sure about the end of string, $, as usually these examples have other tags and strings. Dance in the Vampire Bund {Special Edition} v01-02 (2019) (Digital) (KG Manga).cbz is an example of a very common use case.

Reply:
That's the exact naming convention of MangaPy. I wanted to make sure that was as narrow as possible.

It is unrelated to "Dance in the Vampire Bund v16-17". That example is not an edge case. It's a pretty standard format where the volume explicitely marked with the prefix v. Hence it matches:
<series names><volume prefix><volume number/range>